app/models/plan.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class Plan < ApplicationRecord
+  validates :monthly_income, presence: true, numericality: true
+  validates :name, presence: true, format: { with: /^[a-z]+$/i }
+  has_many :features, dependent: :destroy
+  has_many :subscriptions, dependent: :destroy
```

Comment:
dont use dependent destroy here. If there is an active subscription it should not destroy. Use restrict with exception